### PR TITLE
fix(contact form): correct undefined variable and minor enh

### PR DESCRIPTION
**Pull request by sc979**
## Description

when submitting a form as a non admin user, the $contact_id was never set

**Fixes** # (none)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 19.10.x
- [x] 20.04.x
- [x] 20.10.x
- [x] 21.04.x (master)

### CODE REVIEW

File: www/include/configuration/configObject/contact/formContact.php [hunk 1]
```diff
@@ -114,52 +114,54 @@
     foreach ($tmp as $key => $value) {
         $cct["contact_svNotifOpts"][trim($value)] = 1;
     }
```
Comment (sc979) [inline www/include/configuration/configObject/contact/formContact.php:116]:
```suggestion
    }
    $dbResult->closeCursor();
```

File: www/include/configuration/configObject/contact/formContact.php [hunk 2]
```diff
@@ -170,21 +173,21 @@
...
-$DBRESULT = $pearDB->query("SELECT contact_id, contact_name
-                            FROM contact
-                            WHERE contact_register = '0' $strRestrinction 
-                            ORDER BY contact_name");
-while ($contacts = $DBRESULT->fetchRow()) {
+$dbResult = $pearDB->query(
+    "SELECT contact_id, contact_name
+    FROM contact
+    WHERE contact_register = '0' " . $strRestriction .
+    " ORDER BY contact_name");
```
Comment (sc979) [inline www/include/configuration/configObject/contact/formContact.php:186]:
```suggestion
    " ORDER BY contact_name"
);
```

File: www/include/configuration/configObject/contact/formContact.php [hunk 3]
```diff
@@ -114,52 +114,55 @@
...
+     * Get ACL information for this user
      */
-    $DBRESULT = $pearDB->query("SELECT acl_group_id 
-                                FROM `acl_group_contacts_relations` 
-                                WHERE `contact_contact_id` = '" . intval($contactId) . "'");
-    for ($i = 0; $data = $DBRESULT->fetchRow(); $i++) {
+    $dbResult = $pearDB->query(
+        "SELECT acl_group_id
+        FROM `acl_group_contacts_relations`
+        WHERE `contact_contact_id` = '" . (int) $contactId . "'");
```
Comment (sc979) [inline www/include/configuration/configObject/contact/formContact.php:134]:
```suggestion
        WHERE `contact_contact_id` = '" . (int) $contactId . "'"
    );
```

File: www/include/configuration/configObject/contact/formContact.php [hunk 4]
```diff
@@ -698,7 +701,7 @@ function myReplace()
 
     if ((isset($ret["contact_enable_notifications"]["contact_enable_notifications"])
```
Comment (sc979) [inline www/include/configuration/configObject/contact/formContact.php:703]:
```suggestion
    if (
        (isset($ret["contact_enable_notifications"]["contact_enable_notifications"])
```

**Review by callapa — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Review by loiclau — APPROVED** [review]

**Comment by sc979** [comment]
I don't have time to resolve the issues on the TA... I'll postpone this work and reopen the PR it I can

**Pull request closed without merging**